models/addComment.model.js

```diff
@@ -0,0 +1,44 @@
+const db = require("../db/connection");
+
+exports.insertComment = (newComment, review_id) => {
+  const { username, body } = newComment;
+
+
+  if (isNaN(Number(review_id))) {
+    return Promise.reject({
+      status: 400,
+      msg: `${review_id} is not a valid number`,
+    });
+  }
+
+  return db
+    .query("SELECT * FROM reviews WHERE review_id = $1;", [Number(review_id)])
+    .then(({ rows }) => {
+
+      if (rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: `No review found for review_id ${review_id}`,
+        });
+      }
+
+      return db.query("SELECT * FROM users WHERE username = $1;", [username]);
+    })
+    .then(({ rows }) => {
+
+      if (rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: `Username not found`,
+        });
+      }
```

Comment:
This error handling works fine, but you might find PSQL returns it's own error, which could get caught and passed to the next error handling middleware.